scheds/rust/scx_layered/src/bpf/main.bpf.c

```diff
@@ -788,8 +790,9 @@ static bool keep_running(struct cpu_ctx *cctx, struct task_struct *p)
 	if (!(tctx = lookup_task_ctx(p)) || !(layer = lookup_layer(tctx->layer)))
 		goto no;
 
+	u64 layer_slice_ns = layer->slice_ns > 0 ? layer->slice_ns : slice_ns;
```

Comment:
You may also want to use layer_slice_ns to refill `p->scx.slice` if the task keeps running. Otherwise the task's time slice is automatically refilled to `SCX_SLICE_DFL` if I'm not wrong.

Maybe something like this (untested):
```
diff --git a/scheds/rust/scx_layered/src/bpf/main.bpf.c b/scheds/rust/scx_layered/src/bpf/main.bpf.c
index 14e6fc9..cb9ccbe 100644
--- a/scheds/rust/scx_layered/src/bpf/main.bpf.c
+++ b/scheds/rust/scx_layered/src/bpf/main.bpf.c
@@ -817,6 +817,7 @@ static bool keep_running(struct cpu_ctx *cctx, struct task_struct *p)
 		 */
 		if (disable_topology) {
 			if (!scx_bpf_dsq_nr_queued(layer->idx)) {
+				p->scx.slice = layer_slice_ns;
 				lstat_inc(LSTAT_KEEP, layer, cctx);
 				return true;
 			}
@@ -825,6 +826,7 @@ static bool keep_running(struct cpu_ctx *cctx, struct task_struct *p)
 						   tctx->last_cpu :
 						   bpf_get_smp_processor_id());
 			if (!scx_bpf_dsq_nr_queued(dsq_id)) {
+				p->scx.slice = layer_slice_ns;
 				lstat_inc(LSTAT_KEEP, layer, cctx);
 				return true;
 			}
@@ -850,6 +852,7 @@ static bool keep_running(struct cpu_ctx *cctx, struct task_struct *p)
 		scx_bpf_put_idle_cpumask(idle_cpumask);
 
 		if (has_idle) {
+			p->scx.slice = layer_slice_ns;
 			lstat_inc(LSTAT_KEEP, layer, cctx);
 			return true;
 		}
```

Reply:
I think that worked alright, I spent a bit of time trying tuning different layer slice lengths and it's rather interesting to see how the slice can affect throughput vs latency.